src/server/services/wms/qgswmsgetcapabilities.cpp

```diff
@@ -1600,284 +1531,72 @@ namespace QgsWms
         ExGeoBBoxElement.appendChild( nBoundLatitudeElement );
       }
 
-      if ( !wgs84BoundingRect.isNull() ) //LatLonBoundingBox / Ex_GeographicBounding box is optional
+      QDomElement lastCRSElem = layerElem.lastChildElement( version == QLatin1String( "1.1.1" ) ? "SRS" : "CRS" );
+      if ( !lastCRSElem.isNull() )
       {
-        QDomElement lastCRSElem = layerElem.lastChildElement( version == QLatin1String( "1.1.1" ) ? "SRS" : "CRS" );
-        if ( !lastCRSElem.isNull() )
-        {
-          layerElem.insertAfter( ExGeoBBoxElement, lastCRSElem );
-        }
-        else
-        {
-          layerElem.appendChild( ExGeoBBoxElement );
-        }
-      }
-
-      //In case the number of advertised CRS is constrained
-      if ( !constrainedCrsList.isEmpty() )
-      {
-        for ( int i = constrainedCrsList.size() - 1; i >= 0; --i )
-        {
-          appendLayerBoundingBox( doc, layerElem, layerExtent, layerCRS, constrainedCrsList.at( i ), project );
-        }
+        layerElem.insertAfter( ExGeoBBoxElement, lastCRSElem );
       }
-      else //no crs constraint
+      else
       {
-        for ( const QString &crs : crsList )
-        {
-          appendLayerBoundingBox( doc, layerElem, layerExtent, layerCRS, crs, project );
-        }
+        layerElem.appendChild( ExGeoBBoxElement );
       }
     }
 
-
-    void appendLayerBoundingBox( QDomDocument &doc, QDomElement &layerElem, const QgsRectangle &layerExtent,
-                                 const QgsCoordinateReferenceSystem &layerCRS, const QString &crsText,
-                                 const QgsProject *project )
+    void appendLayerCrsExtents( QDomDocument &doc, QDomElement &layerElem, const QMap<QString, QgsRectangle> crsExtents )
     {
-      if ( layerElem.isNull() )
-      {
-        return;
-      }
-
-      if ( crsText.isEmpty() )
-      {
-        return;
-      }
-
       const QString version = doc.documentElement().attribute( QStringLiteral( "version" ) );
 
-      QgsCoordinateReferenceSystem crs = QgsCoordinateReferenceSystem::fromOgcWmsCrs( crsText );
-
-      //transform the layers native CRS into CRS
-      QgsRectangle crsExtent;
-      if ( !layerExtent.isNull() )
+      for ( const QString &crsText : crsExtents.keys() )
```

Comment:
take a const ref and use std::as_const to iterate over it.

Reply:
Done